### PR TITLE
feat: bump walletconnect packages to latest

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Contributes to https://github.com/shapeshift/web/issues/6488 - doesn't yet fix it it but makes it easier to debug things from a sane base of up-do-date packages.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

contributes to https://github.com/shapeshift/web/issues/6488

Low - ensure WC and WC dApps are still happy against prod

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

-  WC is still happy against prod
-  WC dApps is still happy against prod

### Engineering

- ☝🏽 

### Operations

- ☝🏽

### CODE REVIEW

**Comment by gomesalexandre** [comment]
* **#6553** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6553?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: package.json [hunk 1]
```diff
@@ -116,9 +116,9 @@
     "@visx/responsive": "^2.10.0",
     "@visx/scale": "^3.3.0",
     "@visx/xychart": "^2.12.2",
-    "@walletconnect/core": "^2.10.6",
-    "@walletconnect/utils": "^2.10.6",
-    "@walletconnect/web3wallet": "^1.9.5",
+    "@walletconnect/core": "latest",
+    "@walletconnect/utils": "latest",
+    "@walletconnect/web3wallet": "latest",
```
Comment (gomesalexandre) [inline package.json:121]: no

File: src/context/WalletProvider/WalletConnectV2/components/Connect.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import type { WalletConnectV2HDWallet } from '@shapeshiftoss/hdwallet-walletconnectv2'
+import type EthereumProvider from '@walletconnect/ethereum-provider/dist/types/EthereumProvider'
```
Comment (kaladinlight) [inline src/context/WalletProvider/WalletConnectV2/components/Connect.tsx:1]:
```suggestion
import type EthereumProvider from '@walletconnect/ethereum-provider'
```

File: src/plugins/walletConnectToDapps/types.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import type { ChainId } from '@shapeshiftoss/caip'
 import type { FeeDataKey } from '@shapeshiftoss/chain-adapters'
 import type { PartialRecord } from '@shapeshiftoss/types'
-import type { ICore, SessionTypes, SignClientTypes } from '@walletconnect/types'
+import type { Core as TCore } from '@walletconnect/core/dist/types/core'
```
Comment (kaladinlight) [inline src/plugins/walletConnectToDapps/types.ts:4]:
```suggestion
import type WalletConnectCore from '@walletconnect/core'
```

File: src/plugins/walletConnectToDapps/types.ts [hunk 4]
```diff
@@ -42,14 +43,14 @@ export enum CosmosSigningMethod {
...
-  proposal?: SignClientTypes.EventArguments['session_proposal']
+  proposal?: Web3WalletTypes.EventArguments['session_proposal']
   requestEvent?: SupportedSessionRequest<T>
   requestSession?: SessionTypes.Struct
   request?: Web3WalletTypes.AuthRequest
 }
 
 export type WalletConnectState<T = WalletConnectRequest> = {
-  core?: ICore
+  core?: TCore
```
Comment (kaladinlight) [inline src/plugins/walletConnectToDapps/types.ts:53]:
```suggestion
  core?: WalletConnectCore
```

File: src/plugins/walletConnectToDapps/types.ts [hunk 5]
```diff
@@ -81,7 +82,7 @@ export type WalletConnectAction =
   | {
       type: WalletConnectActionType.INITIALIZE
       payload: {
-        core: ICore
+        core: TCore
```
Comment (kaladinlight) [inline src/plugins/walletConnectToDapps/types.ts:85]:
```suggestion
        core: WalletConnectCore
```

File: src/plugins/walletConnectToDapps/types.ts [hunk 6]
```diff
@@ -1,7 +1,8 @@
 import type { ChainId } from '@shapeshiftoss/caip'
 import type { FeeDataKey } from '@shapeshiftoss/chain-adapters'
 import type { PartialRecord } from '@shapeshiftoss/types'
-import type { ICore, SessionTypes, SignClientTypes } from '@walletconnect/types'
+import type { Core as TCore } from '@walletconnect/core/dist/types/core'
+import type { SessionTypes } from '@walletconnect/types'
 import type { PairingTypes } from '@walletconnect/types/dist/types/core/pairing'
```
Comment (kaladinlight) [inline src/plugins/walletConnectToDapps/types.ts:6]:
```suggestion
import type { PairingTypes, SessionTypes } from '@walletconnect/types'
```

File: src/plugins/walletConnectToDapps/walletUtils.ts [hunk 7]
```diff
@@ -1,11 +1,11 @@
 import { Core } from '@walletconnect/core'
-import type { ICore } from '@walletconnect/types'
+import type { Core as TCore } from '@walletconnect/core/dist/types/core'
```
Comment (kaladinlight) [inline src/plugins/walletConnectToDapps/walletUtils.ts:2]:
```suggestion
import type WalletConnectCore from '@walletconnect/core'
```

File: src/plugins/walletConnectToDapps/walletUtils.ts [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
 import type { IWeb3Wallet } from '@walletconnect/web3wallet'
 import { Web3Wallet } from '@walletconnect/web3wallet'
 import { getConfig } from 'config'
 
 let walletConnectWallet: Promise<IWeb3Wallet>
-let core: ICore
+let core: TCore
```
Comment (kaladinlight) [inline src/plugins/walletConnectToDapps/walletUtils.ts:8]:
```suggestion
let core: WalletConnectCore
```

File: src/context/WalletProvider/WalletConnectV2/components/Connect.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/context/WalletProvider/WalletConnectV2/components/Connect.tsx]: c2eb36f1e58f76c471183aec7b5f1e7b606da7c8

File: src/plugins/walletConnectToDapps/types.ts (hunk 3 above)
Comment (gomesalexandre) [reply src/plugins/walletConnectToDapps/types.ts]: c2eb36f1e58f76c471183aec7b5f1e7b606da7c8

File: src/plugins/walletConnectToDapps/types.ts (hunk 4 above)
Comment (gomesalexandre) [reply src/plugins/walletConnectToDapps/types.ts]: c2eb36f1e58f76c471183aec7b5f1e7b606da7c8

File: src/plugins/walletConnectToDapps/types.ts (hunk 5 above)
Comment (gomesalexandre) [reply src/plugins/walletConnectToDapps/types.ts]: c2eb36f1e58f76c471183aec7b5f1e7b606da7c8

File: src/plugins/walletConnectToDapps/types.ts (hunk 6 above)
Comment (gomesalexandre) [reply src/plugins/walletConnectToDapps/types.ts]: c2eb36f1e58f76c471183aec7b5f1e7b606da7c8

File: src/plugins/walletConnectToDapps/walletUtils.ts (hunk 7 above)
Comment (gomesalexandre) [reply src/plugins/walletConnectToDapps/walletUtils.ts]: c2eb36f1e58f76c471183aec7b5f1e7b606da7c8

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**